src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,69 @@
-export const PeopleFilters = () => {
+import React from 'react';
+import { ChangeEventHandler } from 'react';
+import { SearchLink } from './SearchLink';
+
+export const PeopleFilters = ({
+  onSearchChange,
+  sex = '',
+  centuries = [],
+  query = '',
+}: {
+  onSearchChange: ChangeEventHandler<HTMLInputElement>;
+  sex?: string;
+  centuries?: string[];
+  query?: string;
+}) => {
+  const isActive = (key: string, value: string) =>
+    (key === 'sex' && sex === value) || centuries.includes(value);
+
+  const toggleCenturies = (centuryParam: string) => {
+    // Check if centuryParam is already in centuries
+    const isCenturySelected = centuries.includes(centuryParam);
+  
+    // If no change in centuries, return the current array to avoid re-render
+    if (isCenturySelected && centuries.length === 1) {
+      return { centuries };
+    }
+  
+    // Only create a new array when necessary
+    const newCenturies = isCenturySelected
+      ? centuries.filter(century => century !== centuryParam)
+      : [...centuries, centuryParam];
+  
+    // Return the new state only if it differs from the previous state
+    if (newCenturies.length === centuries.length && isCenturySelected) {
+      return { centuries };
+    }
+  
+    return { centuries: newCenturies };
+  };
```

Comment:
The `toggleCenturies` function returns an object with a `centuries` property. Ensure that the `SearchLink` component can handle this structure, as it might expect a different format for the `params` prop. Consider returning the array directly if that's the expected format.